docs/cookbook/langgraph_vs_mirascope/quickstart.md

```diff
@@ -0,0 +1,390 @@
+#
+# 3. **Modular Design**: Mirascope is modular and extensible, enabling developers to tailor the library to their specific needs. Most dependencies are optional and provider-specific, so you can include only the components you require.
+#
+# 4. **Core Primitives**: The library offers two main components:
+#    - **Call and BasePrompt**: These primitives facilitate interactions with LLMs. Developers can create functions that integrate seamlessly with multiple LLM providers through decorators.
+#
+# 5. **Advanced Functionality**: Mirascope supports features like asynchronous function calls, streaming responses, structured data extraction, custom output parsers, and dynamic variable injection.
+#
+# 6. **Integration with FastAPI**: Mirascope includes decorators for wrapping functions into FastAPI routes, making it easier to deploy applications as web services.
+#
+# 7. **Documentation and Examples**: The project comes with extensive usage documentation and example code to help users quickly understand how to utilize its features effectively.
+#
+# ### Installation
+# To install Mirascope, you can use the following command:
+# ```bash
+# pip install mirascope
+# ```
+# ### Compatibility
+# Mirascope is compatible with Python versions 3.10 to 3.11 (not supporting Python 4.0 and above) and is licensed under the MIT License.
+#
+# ### Summary
+# Mirascope positions itself as a simpler, less cumbersome alternative to other LLM frameworks like LangChain. It focuses on providing essential functionalities without unnecessary complexity, making development enjoyable and productive for software engineers looking to integrate LLMs into their applications.
+```
+
+We have enhanced our chatbot's functionality with several key modifications:
+
+    * Added a `WebSearch` tool to the tools parameter in the `@openai.call` decorator.
+    * Refactored the streaming logic and history management into a new `_step` method.
+    * Updated the history to include tool usage and outputs.
+    * Revised the prompt template to instruct the chatbot on how to utilize the new `WebSearch` tool.
+
+## Part 3: Human-in-the-loop
+
+Let us take a look at how we can slot in human input or approval using Mirascope.
+
+### Permission before using tool
+
+Since we are just writing python code, we don't need to setup a `interrupt_before`. We can simply add a function `_interrupt_before` before our tool call, like so:
+
+```python
+def _interrupt_before(self, tool: openai.OpenAITool) -> openai.OpenAITool | None:
+    """Interrupt before the tool is called. Return the modified tool or None."""
+    if not isinstance(tool, WebSearch):
+        return tool
+    response = input("Do you want to use the tool? (y/n): ")
+    if response.lower() in ["n", "no"]:
+        response = input("Do you want to modify the tool? (y/n): ")
```

Comment:
I think we should also make this more clear that we are updating the `query` argument of the tool, something like `f"Do you want to modify the {tool._name()} tool's query? (y/n): "

update in the code as well